Directory.Build.props

```diff
@@ -0,0 +1,17 @@
+<Project>
+  <!-- Temporarily detect that common props haven't been imported since not every project is using this new logic yet. -->
+  <PropertyGroup Condition="'$(_WasCommonPropsImported)' != 'true'">
```

Comment:
This PR has a lot of csproj files changed. Is this temporary check still needed?

Reply:
I'm not sure, there are a few projects I haven't tackled yet with this PR:

```
build\build.proj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
build\docs.proj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
build\loc.proj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
build\publish.proj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
build\sign.proj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
build\symbols.proj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
build\template.runsettingsproj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
setup\Microsoft.VisualStudio.NuGet.BuildTools.swixproj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
setup\Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
setup\Microsoft.VisualStudio.NuGet.Core.vsmanproj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
tools-local\doc.tasks\doc.tasks.csproj:  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'NuGet.sln'))\build\common.props" />
```

For `build.proj`, I'm going to eventually try and fold its logic into `Directory.Build.targets` so when you build from the root, its doing what `build.proj` used to do.  

I am looking into how to make our swix stuff an SDK-style project as well but might do that in a subsequent PR